### PR TITLE
Add delete cache

### PR DESCRIPTION
**Pull request by openorclose**
delete cache with `cache-delete INDEX`.

will update user docs in separate pr

### CODE REVIEW

File: src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public static final String MESSAGE_SUCCESS = "Bookmark's offline copy successfully deleted!";
+    public static final String MESSAGE_FAILURE =
+            "No offline copy is available to delete!!";
+
+    private Index index;
+
+    /**
+     * Instantiates a new DeleteCache Command.
+     *
+     * @param index the index to delete
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java]:
```suggestion
     * @param index the index of the bookmark whose offline copy should be deleted
```

File: src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            throw new CommandException(Messages.MESSAGE_INVALID_BOOKMARK_DISPLAYED_INDEX);
+        }
+
+        Bookmark bookmarkToDelete = lastShownList.get(index.getZeroBased());
+
+        if (bookmarkToDelete.getCachedCopies().isEmpty()) {
+            throw new CommandException(MESSAGE_FAILURE);
+        }
+
+        Bookmark updatedBookmark = new Bookmark(bookmarkToDelete.getName(), bookmarkToDelete.getUrl(),
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java]: Would it be easier to use `BookmarkBuilder` here?

File: src/main/java/seedu/mark/logic/parser/DeleteCacheCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import static java.util.Objects.requireNonNull;
+import static seedu.mark.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.logic.commands.DeleteCacheCommand;
+import seedu.mark.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new CacheCommand object
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/parser/DeleteCacheCommandParser.java]:
```suggestion
 * Parses input arguments and creates a new DeleteCacheCommand object
```

File: src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java (hunk 2 above)
Comment (openorclose) [reply src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java]: Nope, because bookmark build can only add cached  copies, not delete them

**Pull request merged**